### ISSUE TITLE
Energy analysis for Dynamo installation not working

### ISSUE THREAD

**Issue by KasperLarsen** [tt-acm/EnergyAnalysisForDynamo#110]
When I install the package it simply does not appear as an option for use. The other packages I have installed appeared as intended. If I go to manage packages, it seems to be installed. 

Am I missing something? Perhaps I need to have Revit installed for this to work, even-though it is stated to be a stand alone program?
![image](https://cloud.githubusercontent.com/assets/20505377/16902787/3b01821c-4c69-11e6-8e9c-e128354e3379.png)
![image](https://cloud.githubusercontent.com/assets/20505377/16902788/4ab666c8-4c69-11e6-9ba2-292b38cadf49.png)

-Hope you can help, Kasper

**Comment by tatlin** [comment]
Yes this package uses objects from Revit and requires you to be running Dynamo on top of Revit. You can check which packages did not load via the Dynamo console, this shows what has happened since Dynamo started up in the session.

-matt

**Comment by KasperLarsen** [comment]
It seems that Dynamo cannot load the files needed for Energy analysis. The question is then, how do I get it to do it? I have not been able to find any manuals on the subject, and I am not an experience Revit user. 

Hope you can help!

Copied from the console:

Dynamo log started 18-07-2016 22:11:47
Requesting version update info...
Dynamo -- Build 1.0.0.1180
Duplicate migration type registered for SunPathDirection
Dynamo is up to date.
Dynamo will use the package manager server at : https://www.dynamopackages.com
Build error for library: C:\Users\Kasper\AppData\Roaming\Dynamo\Dynamo Core\1.0\packages\Energy Analysis for  Dynamo\bin\EnergyAnalysisForDynamo.dll
Filen eller assemblyen 'RevitNodes, Version=0.9.1.4062, Culture=neutral, PublicKeyToken=null' eller en af dens afhængigheder kunne ikke indlæses. Den angivne fil blev ikke fundet.
Failed to load library: C:\Users\Kasper\AppData\Roaming\Dynamo\Dynamo Core\1.0\packages\Energy Analysis for  Dynamo\bin\EnergyAnalysisForDynamo.dll
Dynamo.Exceptions.LibraryLoadFailedException: Failed to load library: C:\Users\Kasper\AppData\Roaming\Dynamo\Dynamo Core\1.0\packages\Energy Analysis for  Dynamo\bin\EnergyAnalysisForDynamo.dll
Exception when attempting to load package Energy Analysis for  Dynamo from C:\Users\Kasper\AppData\Roaming\Dynamo\Dynamo Core\1.0\packages\Energy Analysis for  Dynamo
System.Reflection.ReflectionTypeLoadException: En eller flere af de anmodede typer kan ikke indlæses. Hent egenskaben LoaderExceptions for at få flere oplysninger.
DynamoPackageManager (id: FCABC211-D56B-4109-AF18-F434DFE48139) extension is added
Backup files timer is started with an interval of 60000 milliseconds
RENDER : Rendering Tier: 2
RENDER : Pixel Shader 3 Supported: True
RENDER : Pixel Shader 4 Supported: False
RENDER : Software Effect Rendering Supported: True
RENDER : Maximum hardware texture size: 16384;16384
DynamoManipulationExtension (id: 58B0496A-E3F8-43D9-86D2-94823D1D0F98) view extension is added
DynamoPublishExtension (id: BCABC211-D56B-4109-AF18-F434DFE48139) view extension is added
Welcome to Dynamo!
00:00:02.1008067 elapsed for loading Autodesk Dynamo Studio main window.
Failed to load library: Operators
Failed to load library: Operators
Failed to load library: Operators
Failed to load library: Operators
Failed to load library: Operators
Failed to load library: Operators
Failed to load library: Operators
Failed to load library: BuiltIn

**Comment by schaggar16** [comment]
I am having the same issue with Dynamo 1.0, 
![image](https://cloud.githubusercontent.com/assets/20593879/17051077/9242d264-4fee-11e6-8320-bc3cc2aac566.png)

**Comment by KasperLarsen** [comment]
It work like a charm when I opened Dynamo from Revit and installed the Energy analysis, so try it that way schagger16.

**Comment by schaggar16** [comment]
Hi Kasper, I am still having the same issue. which version of Dynamo/Revit/Energy analysis are you using? did you do a clean re-install?

**Comment by KasperLarsen** [comment]
I have had succes with it on multiple versions on Dynamo. Be sure to open Dynamo from Revit. Then install the package. Close and opgen Dynamo again, and it should be there.

**Comment by schaggar16** [comment]
Very strange. Did exactly as you said, but go the same error: System.ArgumentException: An item with the same key has already been added. I have also tried it with various version, no luck. 
Any ideas on what this actually means?

**Comment by schaggar16** [comment]
had a further trial and error session. it seems that the package works with certain templates (it has worked with the default mechanical template but not the architectural template). 
I am having issues getting the UI dropdown boxes to work though, i get the error that the package has not been implemented.

**Comment by KasperLarsen** [comment]
I couldn't even find the dropdown boxes, there is however the setting box you can use instead. It simplifies it somewhat, but you can atleast work further with it

**Comment by RazanRashid** [comment]
This package is not working for Revit 2023. 
In my case the gbXML file could not be uploaded.
